langchain/chains/llm_for_loop.py

```diff
@@ -0,0 +1,63 @@
+"""Chain that first uses an LLM to generate multiple items then loops over them."""
+from typing import Any, Dict, List
+
+from pydantic import BaseModel, Extra, root_validator
+
+from langchain.chains.base import Chain
+from langchain.chains.llm import LLMChain
+from langchain.prompts.base import ListOutputParser
+
+
+class LLMForLoopChain(Chain, BaseModel):
+    """Chain that first uses an LLM to generate multiple items then loops over them."""
+
+    llm_chain: LLMChain
+    """LLM chain to use to generate multiple items."""
+    apply_chain: Chain
```

Comment:
can this be a list of chains? and guarantee that the number of generated items is equal to the length of the list?